taskomatic/jobs/rules_git_sync.py

```diff
@@ -244,6 +244,25 @@ def rule_update(conn, cur, to_update_names, rules_dict):
                     WHERE v.name = ir.name""",
                    to_update, template='(%s, %s, %s, %s, %s, %s, %s, %s::integer, %s::integer, %s::integer, %s::integer, %s::timestamp)',
                    page_size=len(to_update))
+
+    rule_id_mapping = {}
+    cur.execute("""SELECT id, name FROM insights_rule""")
+    for rule_id, rule_name in cur.fetchall():
+        rule_id_mapping[rule_name] = rule_id
+
+    rule_ids = [rule_id_mapping[elem[0]] for elem in to_update]
+
+    cve_rule_tuples = []
+    for rule_id in rule_ids:
+        cur.execute("""SELECT cve.id FROM cve_metadata AS cve
+                JOIN cve_rule_mapping AS crm ON crm.cve_id = cve.id
+                JOIN insights_rule AS ir ON ir.id = crm.rule_id WHERE ir.id = %s""", (rule_id,))
+        cve_rule_tuples.extend([(cve_id[0], rule_id) for cve_id in cur.fetchall()])
+
+    if cve_rule_tuples:
+        execute_values(cur, """INSERT INTO cve_rule_mapping (cve_id, rule_id) VALUES %s ON CONFLICT DO NOTHING""",
+                       cve_rule_tuples, template="(%s, %s)", page_size=len(cve_rule_tuples))
+
```

Comment:
This code looks duplicated, can you move it to a function if it's really duplicated?